qr-generator/src/test/java/com/qpay/qrgenerator/service/impl/QrGeneratorServiceImplTest.java

```diff
@@ -0,0 +1,46 @@
+package com.qpay.qrgenerator.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.client.j2se.MatrixToImageConfig;
+import com.google.zxing.client.j2se.MatrixToImageWriter;
+import com.google.zxing.qrcode.QRCodeWriter;
+import com.qpay.qrgenerator.model.dto.QrCodeCreation;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.ByteArrayOutputStream;
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class QrGeneratorServiceImplTest {
+
+    @Autowired
+    private QrGeneratorServiceImpl qrGeneratorService;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    private static final int QR_CODE_SIZE = 240;
+
+    @Test
+    void should_GenerateQrCode() throws Exception {
```

Comment:
First of all, it is not integration test. This is a unit test where you check only this class. You don't need to start the whole app.

In general, test is useless.
You do not need to repeat your code. What you are doing now is testing qr code library. But it does not make sense. It's already tested by developers of the library.

You can either check that result of this call `qrGeneratorService.generateQrCode(qrCodeCreation);` is not empty or generate beforehand qr code and check it with the result. Second approach is better but it only works if qr codes are same all the time for the same data. If it is not, it won't work.  So, for simplicity check that result is not empty